### ISSUE TITLE
New object

### ISSUE THREAD

**Issue by kuaileqipaoshui** [Open3DA/LL3DA#15]
Excuse me, I have a question, is the prediction of the box(ov-det) invalid for the objects other than the 17 objects defined? Because I found that objects will be filtered here(captioner.py-line394), if there are new categories, will they all be judged as others, so that it is impossible to predict?
![ll3da](https://github.com/Open3DA/LL3DA/assets/137666741/d64b71d9-46e2-459e-a3b7-0a374652611e)
If there's a new object, a fruit, how to predict its position?

**Comment by ch3cook-fdu** [comment]
The last class of `sem_cls_logits` is the `no object` class. Open-vocabulary detection is designed to extend a model’s ability to localize and recognize object beyond a close and pre-defined category set.

**Comment by kuaileqipaoshui** [comment]
> The last class of `sem_cls_logits` is the `no object` class. Open-vocabulary detection is designed to extend a model’s ability to localize and recognize object beyond a close and pre-defined category set.

Is it possible to locate objects in a pre-defined set of classes in ov-det so that the description is generated for these objects only? Wouldn't it be possible to locate a new object without generating a related object description?
If retrain detection, can  increase the object category? Such as adding a fruit category

**Comment by ch3cook-fdu** [comment]
You can just filter and re-label the categories in the generated texts.

**Comment by kuaileqipaoshui** [comment]
> You can just filter and re-label the categories in the generated texts.

I'm sorry, I didn't understand what you said. For example, if I want to detect the location of a banana in a new scene, can it output the location of a banana like the one in the template?
![ll3da1](https://github.com/Open3DA/LL3DA/assets/137666741/26a210e7-3e58-454c-8a1b-02a4c13131e2)
So when I filter, it will think of the bananas as others categories. If I re-label the categories, such as 18:banana, would it be right?
 change self.num_semcls=19?
![ll3da2](https://github.com/Open3DA/LL3DA/assets/137666741/8b783442-5f6a-441e-a759-40d853364f52)

**Comment by ch3cook-fdu** [comment]
If you are looking for a grounding model, you can design input text instructions like “locate the banana”.

**Comment by kuaileqipaoshui** [comment]
> locate the banana



> If you are looking for a grounding model, you can design input text instructions like “locate the banana”.

I just need to have some text instruction like this in eval and it can detect the position of the new object?

**Comment by kuaileqipaoshui** [comment]
> If you are looking for a grounding model, you can design input text instructions like “locate the banana”.

The generated answer gives the center of the box and the length, width and height, so how do I visualize the box?
![屏幕截图 2024-05-02 152431](https://github.com/Open3DA/LL3DA/assets/137666741/5b2cfabb-cef8-4838-96d3-e9b4db791bcc)
How can I reconstructed the 3D box?

**Comment by ch3cook-fdu** [comment]
Please refer to https://github.com/ch3cook-fdu/3d-pc-box-viz for more visualization functions

**Comment by kuaileqipaoshui** [comment]
> Please refer to https://github.com/ch3cook-fdu/3d-pc-box-viz for more visualization functions

I'm sorry to bother you. Could you tell me in more detail how to operate it? It's a little difficult for me.

**Comment by kuaileqipaoshui** [comment]
> Please refer to https://github.com/ch3cook-fdu/3d-pc-box-viz for more visualization functions

It's good to see the results of your work.  Can you explain in detail how to decode 3D box? I try to decode it, but failed. Looking forward to your reply.

**Comment by ch3cook-fdu** [comment]
The code for decoding box coordinates can be found in https://github.com/Open3DA/LL3DA/blob/main/eval_utils/evaluate_ovdet.py#L163-L201. Please refer to https://github.com/ch3cook-fdu/Vote2Cap-DETR/issues/11 for visualization.
